src/System.Management.Automation/engine/parser/TypeInferenceVisitor.cs

```diff
@@ -2490,7 +2485,8 @@ public static bool AstAssignsToSameVariable(this VariableExpressionAst variableA
             if (parameterAst != null)
             {
                 return variableAstVariablePath.IsUnscopedVariable &&
-                       parameterAst.Name.VariablePath.UnqualifiedPath.Equals(variableAstVariablePath.UnqualifiedPath, StringComparison.OrdinalIgnoreCase);
+                       parameterAst.Name.VariablePath.UnqualifiedPath.Equals(variableAstVariablePath.UnqualifiedPath, StringComparison.OrdinalIgnoreCase) &&
+                       parameterAst.Parent.Parent.Extent.EndOffset > variableAst.Extent.StartOffset;
```

Comment:
@MartinGC94 This change came from the commit https://github.com/PowerShell/PowerShell/pull/18691/commits/c91e29f27ef323a52429ae59661692001ddd8348, which was reverted by https://github.com/PowerShell/PowerShell/pull/18691/commits/1289c4e64cdb5de26bce325d36968b771d2bbd5a. However, this change was left behind. Do you mean to revert this change as well?

Reply:
No, that was a fix for this: https://github.com/PowerShell/PowerShell/pull/18691#issuecomment-1332909496 I was just lazy and put 2 changes in 1 commit which was clearly a bad idea 😄